AgentInterfaces/MaintenanceSchedule.cs

```diff
@@ -29,7 +26,15 @@ public MaintenanceSchedule(MaintenanceSchedule other)
         {
             DocumentIncarnation = other.DocumentIncarnation;
             MaintenanceEvents = other.MaintenanceEvents.Select((e) => new MaintenanceEvent(e)).ToList();
-            ReportingVmId = other.ReportingVmId;
+        }
+
+        public MaintenanceSchedule(string eventType, string eventStatus, string eventSource)
```

Comment:
similar as the other comment, if this constructor is used in testing, code should reside in the test project.